### ISSUE TITLE
[iOS 13] Webveiw dismisses when backgrounding host app

### ISSUE THREAD

**Issue by cjwillis809** [AzureAD/microsoft-authentication-library-for-objc#723]
In iOS 13, I'm having an issue where if I background my app while the authentication webview is live, when I come back to the app, the webview window would dismiss. I would then need to re-initiate the login call. After testing, it seems like no matter what if I background the app while the webview is open, it will dismiss when I come back.

I noticed that in the sample app this behavior does not exist. I wanted to see if anyone could point me in the right direction on what I may be doing wrong to cause this weird backgrounding behavior.

Here's how we are instantiating MSAL:
```
   init() {
        MSALGlobalConfig.httpConfig.timeoutIntervalForRequest = 30
        createContext()
    }

    func createContext() {
        guard let authorityURL = URL(string: authURL) else {
            log.error("Unable to get authority URL: \(SignInErrors.authURLParsingError.localizedDescription)")
            return
        }

        // Try and initialize application context.
        do {
            let authority = try MSALAADAuthority(url: authorityURL)
            let configuration = MSALPublicClientApplicationConfig(clientId: clientID, redirectUri: redirectURI, authority: authority)
            applicationContext = try MSALPublicClientApplication(configuration: configuration)
        } catch {
            log.error("Unable to initialize application context")
        }
    }
```
And here's how we are acquiring the token. We have the function `acquireTokenInteractively()` that handles this:
```
private func acquireTokenInteractively(withViewController viewController: UIViewController, completionHandler: @escaping CompletionHandler, errorHandler: @escaping ErrorHandler) {
        guard let applicationContext = applicationContext else {
            errorHandler(SignInErrors.authURLParsingError)
            return
        }
        
        let webViewParameters = MSALWebviewParameters(parentViewController: viewController)
        let interactiveParameters = MSALInteractiveTokenParameters(scopes: [Constants.Scopes.mainScope], webviewParameters: webViewParameters)
        applicationContext.acquireToken(with: interactiveParameters) { (result, error) in
            if let error = error {
                errorHandler(error)
            } else {
                self.handleResult(result: result, completionHandler: completionHandler, errorHandler: errorHandler)
                self.getExtraScopesTokens(withViewController: viewController)
            }
        }
    }
```

Some other info:
- We are using MSAL version `0.7.0`. I also upgraded to `1.0.0` to see if that made a difference but it did not. 
- The parent view controller is being passed in from another function not displayed. The view controller that is passed in is our login view controller where the user taps a button to fire off MSAL authentication. 
Any guidance on this is greatly appreciated. Thanks!

**Comment by oldalton** [comment]
Do you get any error from MSAL when you come back to the app? Is your own controller still there or also dismissed? I'd also recommend you to collect MSAL verbose logs for the next steps. Thanks!

**Comment by cjwillis809** [comment]
Thanks for the response! I receive the following error: `The operation couldn’t be completed. (MSALErrorDomain error -50005.)`. Yes, my view controller still appears after it dismisses. I enabled the verbose logs like so
```
MSALGlobalConfig.loggerConfig.piiEnabled = true
MSALGlobalConfig.loggerConfig.logLevel = .verbose
```
but I don't receive any new information. I only receive the error above.

**Comment by oldalton** [comment]
@cjwillis809, you should receive additional information in MSAL logger callback after setting those two lines. Have you checked the output there?

**Comment by cjwillis809** [comment]
@oldalton Thanks for that guidance, I forgot to set the callback. Here's what I receive:

```
MSAL log: TID=1253058 MSAL 0.7.0 iOS 13.1.2 [2019-10-18 16:11:18 - AB959D8D-BE6C-4BE0-BCBE-0BAD5B653E34] [MSAL] Authorization session was cancelled programatically
MSAL log: TID=1253058 MSAL 0.7.0 iOS 13.1.2 [2019-10-18 16:11:18] Stop background task with type 0
MSAL log: TID=1253058 MSAL 0.7.0 iOS 13.1.2 [2019-10-18 16:11:18 - AB959D8D-BE6C-4BE0-BCBE-0BAD5B653E34] [MSAL] Interactive flow finished result (null), error: -51701 error domain: MSIDErrorDomain
MSAL log: TID=1253058 MSAL 0.7.0 iOS 13.1.2 [2019-10-18 16:11:18 - AB959D8D-BE6C-4BE0-BCBE-0BAD5B653E34] [MSAL] acquireToken returning with error: (MSALErrorDomain, -50005) Authorization session was cancelled programatically.
```

It looks like it's saying the app itself is doing something to dismiss the window. We are doing some things in `viewDidApear()` in the view controller we're using as a parent. Could that be the issue?

**Comment by oldalton** [comment]
Thanks for providing the log. Can you please make sure you're not calling this method in your code? https://azuread.github.io/microsoft-authentication-library-for-objc/Classes/MSALPublicClientApplication.html#/c:objc(cs)MSALPublicClientApplication(cm)cancelCurrentWebAuthSession

**Comment by cjwillis809** [comment]
@oldalton Yes we are calling that method whenever we come into the foreground from being backgrounded. I removed that function call and it resolved the issue!

The reason for setting this, however, is there is behavior where if a user were to lock the phone while the **Sign In** modal is showing (the one with "Cancel" and "Continue"), and then unlocked their phone, they could no longer try to log in again. MSAL gives an error code -42402 which denotes an authentication session is already active and another one cannot be started again. This only seems to happen when locking/unlocking the phone while the modal is up. Once the web view is showing, everything works fine, whether backgrounding or locking/unlocking.

Do you happen to know of a way to resolve this behavior? It seems like we need a way to close the authentication session when you lock the phone and are shown the modal. This is likely an edge case, but it would be inconvenient for users who, accidentally or otherwise, lock their phone when the modal is showing and they can no longer try to login. The only fix we've seen is killing the app and then retrying.

If there is not a built-in solution for this yet, I think we can roll with removing the function call and dealing with the edge case behavior. I did want to see if the above issue I referenced is already solved for however. Thanks for your help!

Just for clarification, here are the reproduction steps:
1. Navigate to the screen where your login action is
2. Initiatie the login process
3. When the "Wants to Use x to Sign In" modal comes up, lock the device (tapping the power button)
4. Unlock the device.
5. Try to initiate login again
Actual: MSAL returns an error saying a session is already started.

**Comment by oldalton** [comment]
@cjwillis809, the behavior you found when device gets locked looks like iOS bug. I was able to reproduce it, and iOS doesn't return as a callback when that happens. Let me report it to Apple. Thanks!
